src/app/operators/dialogs/create-rgb/create-rgb.component.ts

```diff
@@ -0,0 +1,135 @@
+        });
+
+        this.inputLayersubscriptions = this.form.controls['inputLayers'].valueChanges
+            .subscribe((inputLayers: Array<RasterLayer<RasterSymbology>>) => { // set meaningful default values if possible
+                const colors = ['red', 'green', 'blue'];
+                inputLayers.forEach((inputRaster, i) => {
+                    if (inputRaster && !this.form.controls[`${colors[i]}Min`].value) {
+                        this.form.controls[`${colors[i]}Min`].setValue(inputLayers[i].operator.getDataType('value').getMin());
+                    }
+                    if (inputRaster && !this.form.controls[`${colors[i]}Max`].value) {
+                        this.form.controls[`${colors[i]}Max`].setValue(inputLayers[i].operator.getDataType('value').getMax());
+                    }
+                });
+            });
+    }
+
+    ngOnDestroy() {
+        if (this.inputLayersubscriptions) {
+            this.inputLayersubscriptions.unsubscribe();
+        }
+    }
+
+    add() {
+        const inputs: Array<RasterLayer<RasterSymbology>> = this.form.controls['inputLayers'].value;
+        const operators = inputs.map(layer => layer.operator);
+
+        if (inputs.length !== 3) {
+            this.notificationService.error('RGBA calculation requires 3 inputs.');
+            return;
+        }
+        if (unequalProjections(operators)) {
+            this.notificationService.error('Input rasters must be of same projection.');
+            return;
+        }
+
+        const unit = new Unit({
+            interpolation: Interpolation.Unknown,
+            measurement: 'unknown',
+            unit: 'unknown',
+            min: 1,
+            max: 0xffffffff,
+        });
+
+        this.projectService.addLayer(new RasterLayer({
+            name: `RGB of (${inputs.map(layer => layer.name).join(', ')})`,
+            operator: new Operator({
+                operatorType: new RgbaCompositeType({
+                    rasterRedMin: this.form.controls['redMin'].value,
+                    rasterRedMax: this.form.controls['redMax'].value,
+                    rasterRedScale: this.form.controls['redScale'].value,
+                    rasterGreenMin: this.form.controls['greenMin'].value,
+                    rasterGreenMax: this.form.controls['greenMax'].value,
+                    rasterGreenScale: this.form.controls['greenScale'].value,
+                    rasterBlueMin: this.form.controls['blueMin'].value,
+                    rasterBlueMax: this.form.controls['blueMax'].value,
+                    rasterBlueScale: this.form.controls['blueScale'].value,
+                }),
+                projection: operators[0].projection,
+                rasterSources: operators,
+                resultType: ResultTypes.RASTER,
+                attributes: ['value'],
+                dataTypes: new Map<string, DataType>().set('value', DataTypes.UInt32),
```

Comment:
while the datatype is correct, i think we need a dedicated datatype (or maybe a layertype? ) in the future.

Reply:
This is a separate task for a refactoring, since this incorporates way more operators than this one. You could create a task for that.